src/compas_timber/connections/joint.py

```diff
@@ -259,3 +259,126 @@ def _beam_side_incidence(beam_a, beam_b, ignore_ends=True):
             face_angles[face_index] = angle_vectors(face.normal, centerline_vec)
 
         return face_angles
+
+    @staticmethod
+    def _beam_ref_side_incidence(beam_a, beam_b, ignore_ends=True):
+        # compared to beam_side_incidence, this function considers the ref_sides and not faces and forms part of the transition to the new system
+        """Returns a map of ref_side indices of beam_b and the angle of their normal with beam_a's centerline.
+
+        This is used to find a cutting plane when joining the two beams.
+
+        Parameters
+        ----------
+        beam_a : :class:`~compas_timber.parts.Beam`
+            The beam that attaches with one of its ends to the side of beam_b.
+        beam_b : :class:`~compas_timber.parts.Beam`
+            The other beam.
+        ignore_ends : bool, optional
+            If True, only the first four ref_sides of `beam_b` are considered. Otherwise all ref_sides are considered.
+
+        Examples
+        --------
+        >>> ref_side_angles = Joint.beam_side_incidence(beam_a, beam_b)
+        >>> closest_ref_side_index = min(ref_side_angles, key=ref_side_angles.get)
+        >>> cutting_plane = beam_b.ref_sides[closest_ref_side_index]
+
+        Returns
+        -------
+        dict(int, float)
+            A map of ref_side indices of beam_b and their respective angle with beam_a's centerline.
+
```

Comment:
this is a protected methods, so this docstring won't show anywhere.
either remove it or consider making this method public